R/MsBackendTimsTof-functions.R

```diff
@@ -88,4 +88,35 @@
 
 MsBackendTimsTof <- function() {
   new("MsBackendTimsTof")
-}
\ No newline at end of file
+}
+
+#' Read peaks from a single .d folder.
+#'
+#' @param x `character(1)` indicating the file to read from.
+#'
+#' @noRd
+# .timstof_peaks <- function(x) {
+#   tms <- OpenTIMS(x)
+#   tmp <- query(tms, tms@frames$Id, c("frame", "scan", "mz", "intensity"))
+#   # the result below should be a list of matrices (and not data.frames), right?
+#   unname(split.data.frame(as.matrix(tmp[, c("mz", "intensity")]), 
+#                           factor(paste(tmp$frame, tmp$scan))))
+# }
+
+#' Read columns (mz, intensity or both) from a single file
+#' 
+#' @param x `character(1)` indicating the file to read from.
+#'
+#' @noRd
+.read_frame_col <- function(x, columns) {
+  tms <- OpenTIMS(x)
+  if (any(!columns %in% tms@all_columns))
+    stop("Invalid value for columns")
+  if (!length(sd <-setdiff(columns, c("frame", "scan"))))
+    stop("At least one column value different from 'frame' and 'scan' required")
+  tmp <- query(tms, tms@frames$Id, c("frame", "scan", sd))
+  f <- factor(paste(tmp$frame, tmp$scan))
+  if (length(sd) == 1)
+    unname(split(tmp[, sd], f))
+  else unname(split.data.frame(as.matrix(tmp[, sd]), f))
```

Comment:
before returning you'll also need to ensure the data is in the correct order and you need to take care of scans for which nothing was measured/returned. Something like:

```r
tmp <- unname(split(tmp[, sd], f)[paste(indices[, 1], indices[, 2])])
```

could work, but maybe you find a better way.